aiida_common_workflows/workflows/relax_and_bands.py

```diff
@@ -0,0 +1,324 @@
+            serializer=to_aiida_type,
+            help='Reference target distance between neighboring k-points along the path in units 1/Å.',
+        )
+        spec.input(
+            'seekpath_parameters.symprec',
+            valid_type=orm.Float,
+            default=lambda: orm.Float(0.00001),
+            serializer=to_aiida_type,
+            help='The symmetry precision used internally by SPGLIB.',
+        )
+        spec.input(
+            'seekpath_parameters.angle_tolerance',
+            valid_type=orm.Float,
+            default=lambda: orm.Float(-1.0),
+            serializer=to_aiida_type,
+            help='The angle tollerance used internally by SPGLIB.',
+        )
+        spec.input(
+            'seekpath_parameters.threshold',
+            valid_type=orm.Float,
+            default=lambda: orm.Float(0.0000001),
+            serializer=to_aiida_type,
+            help='The treshold for determining edge cases. Meaning is different depending on bravais lattice.',
+        )
+        spec.input(
+            'seekpath_parameters.with_time_reversal',
+            valid_type=orm.Bool,
+            default=lambda: orm.Bool(True),
+            serializer=to_aiida_type,
+            help='If False, and the group has no inversion symmetry, additional lines are returned.',
+        )
+
+        namspac = spec.inputs.create_port_namespace('relax_inputs')
+        namspac.absorb(CommonRelaxInputGenerator.spec().inputs)
+        namspac['protocol'].non_db = True
+        namspac['spin_type'].non_db = True
+        namspac['relax_type'].non_db = True
+        namspac['electronic_type'].non_db = True
+        namspac['magnetization_per_site'].non_db = True
+        namspac['threshold_forces'].non_db = True
+        namspac['threshold_stress'].non_db = True
+        namspac['engines']['relax']['options'].non_db = True
+
+        namspac2 = spec.inputs.create_port_namespace('bands_inputs')
+        namspac2.absorb(CommonBandsInputGenerator.spec().inputs, exclude=('parent_folder'))
+        namspac2['engines']['bands']['options'].non_db = True
+        namspac2['bands_kpoints'].required = False
+
+        namspac3 = spec.inputs.create_port_namespace('second_relax_inputs')
+        namspac3.absorb(CommonRelaxInputGenerator.spec().inputs, exclude=('structure'))
+        namspac3['protocol'].non_db = True
+        namspac3['spin_type'].non_db = True
+        namspac3['relax_type'].non_db = True
+        namspac3['electronic_type'].non_db = True
+        namspac3['magnetization_per_site'].non_db = True
+        namspac3['threshold_forces'].non_db = True
+        namspac3['threshold_stress'].non_db = True
+        namspac3['engines']['relax']['options'].non_db = True
+        for key in namspac3:
+            namspac3[key].required = False
+            namspac3[key].populate_defaults = False
+            namspac3[key].default = ()
+        namspac3['relax_type'].required = True
+        namspac3['relax_type'].default = RelaxType.NONE
```

Comment:
Wouldn't rely on this through a default because then a user can run anything else. We should always set this no? So just set it in the workchain step when preparing the inputs

Reply:
I thought this was against the idea to fully expose the inputs of the sub-processes, but ok, this makes sense. And also avoids the problem of my last comments above.